lib/column/date32.go

```diff
@@ -171,6 +214,19 @@ func (col *Date32) AppendRow(v interface{}) error {
 	return nil
 }
 
+func (col *Date32) parseDate(str string) (datetime time.Time, err error) {
+	defer func() {
+		if err == nil {
+			err = dateOverflow(minDate32, maxDate32, datetime, defaultDateFormat)
+		}
+	}()
+	datetime, err = time.Parse(defaultDateFormat, str)
+	if err == nil {
+		return datetime, nil
+	}
+	return time.Parse(defaultDateTimeFormat, str)
```

Comment:
same comment as above i.e.

any reason we need to call this twice? can't we just return the err and a time.Time{}?